### DISCUSSION TITLE
don't know where to put feed raw content

### DISCUSSION THREAD

**Discussion by lucnap** [amzn/selling-partner-api-models#3464] [General]
this is a snippet of code I'm trying to use to upload some csv data with feed API but can't understand where to put my content data to be uploaded.

```
$contentToUpload =  "sku\tprice\tquantity\r\n" .
            "5200\t16.38\t1\r\n";

$specification  = new CreateFeedDocumentSpecification;
$specification->setContentType('text/tab-separated-values;charset=UTF-8');

$feedDocumentInfo   = $sdk->feeds()->createFeedDocument($accessToken, Regions::EUROPE, $specification);
$feedDocId          = $feedDocumentInfo->getFeedDocumentId();

$feedSpecification = new CreateFeedSpecification([
    'feed_type' => "POST_FLAT_FILE_PRICEANDQUANTITYONLY_UPDATE_DATA",
    "marketplace_ids" => [Marketplace::IT()->id()],
    "input_feed_document_id" => $feedDocId,
]); 

$res = $sdk->feeds()->createFeed($accessToken, Regions::EUROPE, $feedSpecification);

echo $res;

```

**Comment by littleylv** [comment]
[createFeedDocument](https://developer-docs.amazon.com/sp-api/docs/feeds-api-v2021-06-30-reference#createfeeddocument) return [feedDocumentId and **url**](https://developer-docs.amazon.com/sp-api/docs/feeds-api-v2021-06-30-reference#createfeeddocumentresponse). `put` you content to the `url`.

see use case guide [here](https://developer-docs.amazon.com/sp-api/docs/feeds-api-v2021-06-30-use-case-guide#step-3-upload-the-feed-data)

**Comment by lucnap** [comment]
as suggested by littleylv the solution is simple to upload the file to the url given by createFeedDocument response, here it is an example:

        $feedDocumentInfo   = $sdk->feeds()->createFeedDocument($accessToken, Regions::EUROPE, $specification);
        $feedDocId          = $feedDocumentInfo->getFeedDocumentId();

	$ch = curl_init();
	curl_setopt($ch, CURLOPT_URL, $feedDocumentInfo->getUrl());
	curl_setopt($ch, CURLOPT_HTTPHEADER, Array("Content-Type: text/tab-separated-values;charset=UTF-8"));
	curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
	curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "PUT");
	curl_setopt($ch, CURLOPT_POSTFIELDS, $contentToUpload);
	
	$data = curl_exec($ch);
	curl_close($ch);

**Comment by tomaszhanc** [comment]
For PHP I can recommend this [SDK](https://github.com/amazon-php/sp-api-sdk). It's not an official SDK, but it's autogenerated quickly from models provided by Amazon.
